tensorboard/data_compat.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utilities to translate legacy protos to their modern equivalents."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+from tensorboard.plugins.histogram import metadata as histogram_metadata
+
+
+def translate_value(value):
+  """Convert `value` to a new-style value, if necessary and possible.
+
+  An "old-style" value is a value that uses any `value` field other than
+  the `tensor` field. A "new-style" value is a value that uses the
+  `tensor` field. TensorBoard continues to support old-style values on
+  disk; this method converts them to new-style values so that further
+  code need only deal with one data format.
+
+  Arguments:
+    value: A `tf.Summary.Value` object. This argument is not modified.
+
+  Returns:
+    If the `value` is an old-style value for which there is a new-style
+    equivalent, the result is the new-style value. Otherwise---if the
+    value is already new-style or does not yet have a new-style
+    equivalent---the value will be returned unchanged.
+  """
+  handler = {
+      'histo': _translate_histogram_value,
+  }.get(value.WhichOneof('value'))
```

Comment:
I opted not to do that because the definition of such a dict would need to come after all the definitions of each individual function. In my opinion, the file is best read in the way it is currently laid out. Let me know if you have strong opinions to the contrary.

(I also considered `_VALUE_FIELD_MIGRATORS = {}; def _handler(field): def decorator(fn): assert field not in _VALUE_FIELD_MIGRATORS; _VALUE_FIELD_MIGRATORS[field] = fn; return fn`, but this seemed like far too much complexity when a simple inline dict suffices. I'm sure that the performance penalty is fully negligible.)